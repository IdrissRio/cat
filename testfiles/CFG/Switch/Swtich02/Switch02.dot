digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "MethodDecl[28]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Entry[29]"; "Modifiers[30]"; "PrimitiveTypeAccess[32]"; "List[33]"; "List[34]"; "Opt[35]"; "Exit[36]"}
{ rank = same; "List[31]"; "Block[36]"}
{ rank = same; "List[37]"}
{ rank = same; "VarDeclStmt[38]"; "SwitchStmt[47]"; "ExprStmt[57]"}
{ rank = same; "Modifiers[39]"; "PrimitiveTypeAccess[41]"; "List[42]"; "VarAccess[48]"; "Block[49]"; "AssignSimpleExpr[58]"}
{ rank = same; "List[40]"; "VariableDeclarator[43]"; "List[50]"; "VarAccess[59]"; "IntegerLiteral[60]"}
{ rank = same; "List[44]"; "Opt[45]"; "ConstCase[51]"; "ExprStmt[53]"}
{ rank = same; "IntegerLiteral[46]"; "IntegerLiteral[52]"; "AssignSimpleExpr[54]"}
{ rank = same; "VarAccess[55]"; "IntegerLiteral[56]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < Switch02 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[28]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[29]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[30]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[32]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[35]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[36]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[38]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[39]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[41]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[43]" [label="VariableDeclarator\n < a >", fillcolor=white   style=filled  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[45]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[46]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"SwitchStmt[47]" [label="SwitchStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[48]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"Block[49]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstCase[51]" [label="ConstCase\n < 6 >", fillcolor=white   style=filled  ]
"IntegerLiteral[52]" [label="IntegerLiteral\n < 6 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[53]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[54]" [label="AssignSimpleExpr\n < a >", fillcolor=white   style=filled  ]
"VarAccess[55]" [label="VarAccess\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[56]" [label="IntegerLiteral\n < 7 >", fillcolor=white   style=filled  ]
"ExprStmt[57]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[58]" [label="AssignSimpleExpr\n < a >", fillcolor=white   style=filled  ]
"VarAccess[59]" [label="VarAccess\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[60]" [label="IntegerLiteral\n < 8 >", fillcolor=white   style=filled  ]
"Exit[36]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Entry[29]" [style=dashed, color=gray]
"Modifiers[30]" -> "List[31]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Modifiers[30]" [style=dashed, color=gray]
"MethodDecl[28]" -> "PrimitiveTypeAccess[32]" [style=dashed, color=gray]
"MethodDecl[28]" -> "List[33]" [style=dashed, color=gray]
"MethodDecl[28]" -> "List[34]" [style=dashed, color=gray]
"Modifiers[39]" -> "List[40]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "Modifiers[39]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "PrimitiveTypeAccess[41]" [style=dashed, color=gray]
"VariableDeclarator[43]" -> "List[44]" [style=dashed, color=gray]
"Opt[45]" -> "IntegerLiteral[46]" [style=dashed, color=gray]
"VariableDeclarator[43]" -> "Opt[45]" [style=dashed, color=gray]
"List[42]" -> "VariableDeclarator[43]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "List[42]" [style=dashed, color=gray]
"List[37]" -> "VarDeclStmt[38]" [style=dashed, color=gray]
"SwitchStmt[47]" -> "VarAccess[48]" [style=dashed, color=gray]
"ConstCase[51]" -> "IntegerLiteral[52]" [style=dashed, color=gray]
"List[50]" -> "ConstCase[51]" [style=dashed, color=gray]
"AssignSimpleExpr[54]" -> "VarAccess[55]" [style=dashed, color=gray]
"AssignSimpleExpr[54]" -> "IntegerLiteral[56]" [style=dashed, color=gray]
"ExprStmt[53]" -> "AssignSimpleExpr[54]" [style=dashed, color=gray]
"List[50]" -> "ExprStmt[53]" [style=dashed, color=gray]
"Block[49]" -> "List[50]" [style=dashed, color=gray]
"SwitchStmt[47]" -> "Block[49]" [style=dashed, color=gray]
"List[37]" -> "SwitchStmt[47]" [style=dashed, color=gray]
"AssignSimpleExpr[58]" -> "VarAccess[59]" [style=dashed, color=gray]
"AssignSimpleExpr[58]" -> "IntegerLiteral[60]" [style=dashed, color=gray]
"ExprStmt[57]" -> "AssignSimpleExpr[58]" [style=dashed, color=gray]
"List[37]" -> "ExprStmt[57]" [style=dashed, color=gray]
"Block[36]" -> "List[37]" [style=dashed, color=gray]
"Opt[35]" -> "Block[36]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Opt[35]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Exit[36]" [style=dashed, color=gray]
"List[27]" -> "MethodDecl[28]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"Exit[36]" -> "AssignSimpleExpr[58]" [color=red, constraint=false]
"AssignSimpleExpr[58]" -> "Exit[36]" [color=blue, constraint=false, xlabel=" "] 
"AssignSimpleExpr[58]" -> "IntegerLiteral[60]" [color=red, constraint=false]
"IntegerLiteral[60]" -> "AssignSimpleExpr[58]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[60]" -> "ConstCase[51]" [color=red, constraint=false]
"IntegerLiteral[60]" -> "AssignSimpleExpr[54]" [color=red, constraint=false]
"AssignSimpleExpr[54]" -> "IntegerLiteral[60]" [color=blue, constraint=false, xlabel=" "] 
"AssignSimpleExpr[54]" -> "IntegerLiteral[56]" [color=red, constraint=false]
"IntegerLiteral[56]" -> "AssignSimpleExpr[54]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[56]" -> "ConstCase[51]" [color=red, constraint=false]
"ConstCase[51]" -> "IntegerLiteral[56]" [color=blue, constraint=false, xlabel="T "] 
"ConstCase[51]" -> "IntegerLiteral[60]" [color=blue, constraint=false, xlabel="F "] 
"ConstCase[51]" -> "VarAccess[48]" [color=red, constraint=false]
"VarAccess[48]" -> "ConstCase[51]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[48]" -> "VariableDeclarator[43]" [color=red, constraint=false]
"VariableDeclarator[43]" -> "VarAccess[48]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[43]" -> "IntegerLiteral[46]" [color=red, constraint=false]
"IntegerLiteral[46]" -> "VariableDeclarator[43]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[46]" -> "Entry[29]" [color=red, constraint=false]
"Entry[29]" -> "IntegerLiteral[46]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"Entry[29]" -> "Modifiers[30]" -> "PrimitiveTypeAccess[32]" -> "List[33]" -> "List[34]" -> "Opt[35]" -> "Exit[36]" [style=invis]
"VarDeclStmt[38]" -> "SwitchStmt[47]" -> "ExprStmt[57]" [style=invis]
"Modifiers[39]" -> "PrimitiveTypeAccess[41]" -> "List[42]" [style=invis]
"List[44]" -> "Opt[45]" [style=invis]
"VarAccess[48]" -> "Block[49]" [style=invis]
"ConstCase[51]" -> "ExprStmt[53]" [style=invis]
"VarAccess[55]" -> "IntegerLiteral[56]" [style=invis]
"VarAccess[59]" -> "IntegerLiteral[60]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
52", fillcolor=1]
Edges [label="Total Nr. Edges
75", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
12", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
12", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}