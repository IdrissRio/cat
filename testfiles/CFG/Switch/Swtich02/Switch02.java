public class Switch02 {
  void m() {
    int a = 0;
    switch (a) {
    case 6:
      a = 7;
    }
    a = 8;
  }
}