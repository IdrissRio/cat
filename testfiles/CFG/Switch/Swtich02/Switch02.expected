NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [IntegerLiteral[46]]
NodeName: IntegerLiteral[46] Successors: [VariableDeclarator[43]]
NodeName: VariableDeclarator[43] Successors: [VarAccess[48]]
NodeName: VarAccess[48] Successors: [ConstCase[51]]
NodeName: ConstCase[51] Successors: [IntegerLiteral[56], IntegerLiteral[60]]
NodeName: IntegerLiteral[56] Successors: [AssignSimpleExpr[54]]
NodeName: AssignSimpleExpr[54] Successors: [IntegerLiteral[60]]
NodeName: IntegerLiteral[60] Successors: [AssignSimpleExpr[58]]
NodeName: AssignSimpleExpr[58] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [AssignSimpleExpr[58]]
NodeName: AssignSimpleExpr[58] Predecessor: [IntegerLiteral[60]]
NodeName: IntegerLiteral[60] Predecessor: [ConstCase[51], AssignSimpleExpr[54]]
NodeName: ConstCase[51] Predecessor: [VarAccess[48]]
NodeName: VarAccess[48] Predecessor: [VariableDeclarator[43]]
NodeName: VariableDeclarator[43] Predecessor: [IntegerLiteral[46]]
NodeName: IntegerLiteral[46] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: AssignSimpleExpr[54] Predecessor: [IntegerLiteral[56]]
NodeName: IntegerLiteral[56] Predecessor: [ConstCase[51]]
